### DISCUSSION TITLE
Login problem

### DISCUSSION THREAD

**Discussion by raylight75** [spatie/laravel-permission#2068] [Q&A]
After updating from 3 to 5, we added the teams feature but we can't log in. We run everything ok and have set team_id to 1 in most of the tables. Can you provide some settings for Laravel, or for Kernel.php

**Comment by erikn69** [comment]
What exception do you have? 
You need this for login:
https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions#content-working-with-teams-permissions
`team_id` must be on login form(maybe html select), and on `session/cookie` after authentication(for the middleware)
Maybe helps #1906

**Reply by erikn69** [reply]
>We do that but every time your query search for team_id null, if we set team_id to null everything works fine we get permission. We set in session team_id and if we have team_id for example 1 in database but we can't get permission

It means you are doing it wrong, everything is tested, it's hard to tell what you are doing wrong without your implementation code, 

>every time your query search for team_id null

but I'm really sure the problem is where I mentioned, (login, middleware, kernel)
maybe you don´t read https://github.com/spatie/laravel-permission/discussions/1906

**Reply by erikn69** [reply]
Maybe this helps
#2075

**Reply by raylight75** [reply]
Thank you @erikn69

**Comment by raylight75** [comment]
We do that but every time your query search for team_id null, if we set team_id to null everything works fine we get permission. We set in session team_id and if we have team_id for example 1 in database but we can't get permission
